sky/unit/test/widget/global_key_test.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:sky/widgets.dart';
+import 'package:test/test.dart';
+
+import 'build_utils.dart';
+
+void main() {
+  test('Global keys notify add and remove', () {
+    GlobalKey globalKey = new GlobalKey();
+    Container container;
+
+    bool syncListenerCalled = false;
+    bool removeListenerCalled = false;
+
+    void syncListener(GlobalKey key, Widget widget) {
+      syncListenerCalled = true;
+      expect(key, equals(globalKey));
+      expect(container, isNotNull);
+      expect(widget, equals(container));
+    }
+
+    void removeListener(GlobalKey key) {
+      removeListenerCalled = true;
+      expect(key, equals(globalKey));
+    }
+
+    WidgetTester tester = new WidgetTester();
+
+    GlobalKey.registerSyncListener(globalKey, syncListener);
+    GlobalKey.registerRemoveListener(globalKey, removeListener);
+    tester.pumpFrame(0.0, () {
+      container = new Container(key: globalKey);
+      return container;
+    });
+    expect(syncListenerCalled, isTrue);
+    expect(removeListenerCalled, isFalse);
+
+    syncListenerCalled = false;
+    removeListenerCalled = false;
+    tester.pumpFrame(1.0, () => new Container());
+    expect(syncListenerCalled, isFalse);
+    expect(removeListenerCalled, isTrue);
+
+    syncListenerCalled = false;
+    removeListenerCalled = false;
+    GlobalKey.unregisterSyncListener(globalKey, syncListener);
+    GlobalKey.unregisterRemoveListener(globalKey, removeListener);
+    tester.pumpFrame(2.0, () {
+      container = new Container(key: globalKey);
+      return container;
+    });
+    expect(syncListenerCalled, isFalse);
+    expect(removeListenerCalled, isFalse);
+
+    tester.pumpFrame(3.0, () => new Container());
+    expect(syncListenerCalled, isFalse);
+    expect(removeListenerCalled, isFalse);
+  });
```

Comment:
would be good to add a test that has two containers, with one having the key, then have the next test have the same structure but have the key move to the other one, then go back to the first again. Make sure there's no removal listeners called, that the sync one is called three times, that there's no "duplicate" asserts fired, etc.